deposit/bytecode-verification/deposit-spec.ini

```diff
@@ -407,181 +415,98 @@ storage: M
-    [ 192 /* scratchpad      */ := #buf(32, 2) ]                              /* zero_hashes storage index = 2 */
-    [ 512 /* = 320 + 32 *  6 */ := #buf(32, {ZERO_HASHES_0}) ]                /* zero_hashes[0] */
-    [ 448 /* = 320 + 32 *  4 */ := #buf(32, 64) ]                             /* size of sha256 input */
-    [ 192 /* scratchpad      */ := #buf(32, {NODE_1}) ]                       /* sha256 return value via sha256(480, 64) */
-    [ 352 /* = 320 + 32 *  1 */ := #buf(32, {NODE_1}) ]                       /* update node = sha256(node, zero_hashes[0]) */
-    [ 384 /* = 320 + 32 *  2 */ := #buf(32, DEPOSIT_COUNT /Int 2) ]           /* update size */
-    [ 416 /* = 320 + 32 *  3 */ := #buf(32, 1) ]                              /* update height */
-ZERO_HASHES_0: select(M, #hashedLocation({COMPILER}, {ZERO_HASHES}, 0))
+GAS_COST: 1728
+MEM_COST: 672
+
+[get_deposit_root-success-loop1-else]
 NODE_1: #sha256(#buf(32, 0) ++ #buf(32, {ZERO_HASHES_0}))
+ZERO_HASHES_0: select(M, #hashedLocation({COMPILER}, {ZERO_HASHES}, 0))
 +requires:
     // conditions
     andBool DEPOSIT_COUNT &Int 1 =/=Int 1
-; gas usage = 1744 = 1693 + ( 51 = 3*n + n^2 / 512 where n = 17 = 544 / 32 )
-gas:         #symGas(G, 0 => {GAS_COST}, 0 => {GAS_COST}, .List, Cmem({SCHEDULE}, {MEMORY_USED}))
-memory_used: #symMem(448 => 544, .Set)
-GAS_COST: 1318
+GAS_COST: 1718
+MEM_COST: 544
 
-[get_deposit_root-loop]
-; MEMORY_USED_LOOPHEAD: 17 or 21
-LOCAL_MEM_LOOPHEAD: {LOCAL_MEM_BEGIN}
+[get_deposit_root-success-loop]
+LOCAL_MEM_LOOPHEAD: MEM
+    {VYPER_GENERATED_BOUNDS}
     ; locals
-    [ 320 /* = 320 + 32 *  0 */ := #buf(32, 0) ]          /* zero_bytes32: bytes32 = 0x0 */
-    [ 352 /* = 320 + 32 *  1 */ := #buf(32, NODE) ]       /* node: bytes32 */
-    [ 384 /* = 320 + 32 *  2 */ := #buf(32, SIZE) ]       /* size: uint256 */
-    [ 416 /* = 320 + 32 *  3 */ := #buf(32, HEIGHT) ]     /* height */
-    ; garbages
-    [ 192 /* scratchpad      */ := #buf(32, ANON_1) ]     /* sha256 return value */
-    // for if-then branch
-    [ 576 /* = 320 + 32 *  8 */ := #buf(32, ANON_2) ]     /* size of sha256 input */
-    [ 608 /* = 320 + 32 *  9 */ := #buf(32, ANON_3) ]     /* branch[height] */
-    [ 640 /* = 320 + 32 * 10 */ := #buf(32, ANON_4) ]     /* node */
-    // for if-else branch
-    [ 448 /* = 320 + 32 *  4 */ := #buf(32, ANON_5) ]     /* size of sha256 input */
-    [ 480 /* = 320 + 32 *  5 */ := #buf(32, ANON_6) ]     /* node */
-    [ 512 /* = 320 + 32 *  6 */ := #buf(32, ANON_7) ]     /* zero_hashes[height] */
+    [ 320 /* = 320 + 32 * 0 */ := #buf(32, 0) ]          /* zero_bytes32: bytes32 = 0x0 */
+    [ 352 /* = 320 + 32 * 1 */ := #buf(32, NODE) ]       /* node: bytes32 */
+    [ 384 /* = 320 + 32 * 2 */ := #buf(32, SIZE) ]       /* size: uint256 */
+    [ 416 /* = 320 + 32 * 3 */ := #buf(32, HEIGHT) ]     /* height */
 +requires:
     // conditions
     andBool #range(0 <= HEIGHT <= 32)
```

Comment:
Is there a spec that ensures that this reverts if this is not fullfilled? Perhaps `get_deposit_root-revert` can be extended to deal with this case as well

Reply:
Indeed this is a loop invariant that can be proved (using the refinement approach mentioned above), thus it is not possible to violate this. If this is violated somehow (e.g., due to a VM bug), it does not revert but ends up an infinite loop (because the loop termination condition is `height =/= 32` not `height > 32`).

Reply:
I clarified this in the spec: https://github.com/runtimeverification/verified-smart-contracts/pull/320/commits/64b6810fab85582628887fb78b18a745e1e3a131